### PR TITLE
Add braces around blocks.

### PR DESCRIPTION
**Pull request by corycrean**
- Add braces around blocks.
- Add a check for braces to clang-tidy.

### Description

This PR addresses https://github.com/ros-planning/moveit2/issues/854.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_core/collision_detection/src/collision_matrix.cpp [hunk 1]
```diff
@@ -213,19 +217,23 @@ void AllowedCollisionMatrix::setEntry(const std::string& name, bool allowed)
...
     }
+  }
 }
 
 void AllowedCollisionMatrix::setEntry(bool allowed)
 {
   const AllowedCollision::Type v = allowed ? AllowedCollision::ALWAYS : AllowedCollision::NEVER;
   for (auto& entry : entries_)
+  {
     for (auto& it2 : entry.second)
```
Comment (JafarAbdi) [inline moveit_core/collision_detection/src/collision_matrix.cpp:234]: I think you need to run it once more to add the braces to the inner for-loop

Comment (corycrean) [reply moveit_core/collision_detection/src/collision_matrix.cpp:268]: Thanks for catching that.  I re-ran it, hopefully it got them all this time.

**Review by henningkayser — APPROVED** [review]
Great readibility improvement! I think in theory there is nothing that could really go wrong, but this is really a lot to review

**Review by jliukkonen — APPROVED** [review]
Looks good to me.

**Comment by tylerjw** [comment]
Wow, this is a huge change. Thank you for doing this work. I'm happy to merge this if you don't mind rebasing so it can be applied.

**Comment by tylerjw** [comment]
@corycrean I'm sorry this got so out of date and has so many conflicts. Do you think you could rebase this and make it so it could be merged?

File: .clang-tidy [hunk 2]
```diff
@@ -11,6 +11,7 @@ Checks:          '-*,
                   modernize-make-unique,
                   modernize-avoid-bind,
                   misc-unused-parameters,
+		  readability-braces-around-statements,
```
Comment (tylerjw) [inline .clang-tidy:14]: Can you fix the formatting of this so it lines up with the other clang-tidy checks?

File: moveit_planners/test_configs/prbt_ikfast_manipulator_plugin/src/prbt_manipulator_ikfast_moveit_plugin.cpp [hunk 3]
```diff
@@ -1049,9 +1052,10 @@ bool IKFastKinematicsPlugin::searchPositionIK(const geometry_msgs::msg::Pose& ik
                 best_solution = solution;
               }
             }
-            else
+            else {
               // Return first feasible solution
               return true;
+}
```
Comment (tylerjw) [inline moveit_planners/test_configs/prbt_ikfast_manipulator_plugin/src/prbt_manipulator_ikfast_moveit_plugin.cpp:1058]: If you run pre-commit does the formatting of this get fixed?

**Review by tylerjw — APPROVED** [review]

File: moveit_planners/test_configs/prbt_ikfast_manipulator_plugin/src/prbt_manipulator_ikfast_moveit_plugin.cpp (hunk 3 above)
Comment (corycrean) [reply moveit_planners/test_configs/prbt_ikfast_manipulator_plugin/src/prbt_manipulator_ikfast_moveit_plugin.cpp]: Unfortunately no.  But I just fixed this manually.

File: .clang-tidy [hunk 4]
```diff
@@ -11,6 +11,7 @@ Checks:          '-*,
                   modernize-make-unique,
                   modernize-avoid-bind,
                   misc-unused-parameters,
+		              readability-braces-around-statements,
```
Comment (ChrisThrasher) [inline .clang-tidy:14]: Looks like a formatting issue got through in this PR

**Comment by ChrisThrasher** [comment]
I wanted to point out how the argument that drove the creation of #854 only applies to codebases without autoformatters. Misleading indentation is a real problem but it's not a problem MoveIt has so there was no need to add this check to address that. We already have no misleading indentation.

**Pull request merged**